File: c++/4 kyu Range Extraction.cpp
#include <string>
#include <vector>
using namespace std;
std::string range_extraction(std::vector<int> v) {
  stringstream ss("");
  int c=0;
  for(int i=0;i<v.size();i++){
  	if(v[i+2]==v[i]+2){
  	ss<<v[i];
		int j=i+3;
	  while(v[j]-v[j-1]==1){j++;}
		i=j-1;
		ss<<"-"<<v[i]<<",";
	}
	else ss<<v[i]<<",";
	}
  return (ss.str()).substr(0,ss.str().length()-1);
}

File: Kotlin/4 kyu Range Extraction.kt
fun rangeExtraction(a: IntArray): String {
  var s = ""
  var i=0;
  while(i < a.size){
    if(i+2>=a.size) s+=a[i].toString()+","
    else if(a[i+2]==a[i]+2){
      s+=a[i].toString()+"-";
      while(i<a.size-1 && a[i+1]-a[i]==1) i++;
      s+=a[i].toString()+","
    }
    else s+=a[i].toString()+","
    i++;
    }
  return s.dropLast(1)
}
